packages/trees/frontend/sirius-components-trees/src/treeitems/TreeItemDirectEditInput.tsx

```diff
@@ -0,0 +1,164 @@
+/*******************************************************************************
+ * Copyright (c) 2023 Obeo.
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Obeo - initial API and implementation
+ *******************************************************************************/
+import { gql, useMutation, useQuery } from '@apollo/client';
+import TextField from '@material-ui/core/TextField';
+import { useEffect, useState } from 'react';
+import {
+  GQLInitialDirectEditElementLabelData,
+  GQLInitialDirectEditElementLabelInput,
+  TreeItemDirectEditInputProps,
+} from './TreeItemDirectEditInput.types';
+
+const renameTreeItemMutation = gql`
+  mutation renameTreeItem($input: RenameTreeItemInput!) {
+    renameTreeItem(input: $input) {
+      __typename
+      ... on ErrorPayload {
+        message
+      }
+    }
+  }
+`;
+
+const initialDirectEditElementLabeQuery = gql`
+  query initialDirectEditElementLabel($editingContextId: ID!, $representationId: ID!, $treeItemId: ID!) {
+    viewer {
+      editingContext(editingContextId: $editingContextId) {
+        representation(representationId: $representationId) {
+          description {
+            ... on TreeDescription {
+              initialDirectEditTreeItemLabel(treeItemId: $treeItemId)
+            }
+          }
+        }
+      }
+    }
+  }
+`;
+
+export const TreeItemDirectEditInput = ({
+  editingContextId,
+  treeId,
+  treeItemId,
+  refParent,
+  editingkey,
+  onClose,
+}: TreeItemDirectEditInputProps) => {
+  const [state, setState] = useState({
+    newLabel: '',
+    editingkey: editingkey,
+  });
+
+  const {
+    loading: initialLabelTreeItemLoading,
+    data: initialLabelTreeItemItemData,
+    error: initialLabelTreeItemItemError,
+  } = useQuery<GQLInitialDirectEditElementLabelData, GQLInitialDirectEditElementLabelInput>(
+    initialDirectEditElementLabeQuery,
+    {
+      variables: {
+        editingContextId: editingContextId,
+        representationId: treeId,
+        treeItemId: treeItemId,
+      },
+    }
+  );
+
+  useEffect(() => {
+    if (
+      !initialLabelTreeItemLoading &&
+      !initialLabelTreeItemItemError &&
```

Comment:
- Only test if you have some data, no need for the loading value here
- Display an error if there is an error (and don't forget that you can have both an error and some data)